src/makeCalculator.js

```diff
@@ -4,7 +4,43 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculate = {
+    result: 0,
+
+    add(number) {
+      this.result += number;
+    },
+
+    subtract(number) {
+      this.result -= number;
+    },
+
+    multiply(number) {
+      this.result *= number;
+    },
+
+    divide(number) {
+      this.result /= number;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(callback, number) {
+      const isCalculatorMethod = Object.keys(this).some((method) => method === callback);
```

Comment:
The check `Object.keys(this).some((method) => method === callback)` is unnecessary and incorrect. The `callback` parameter is expected to be a function, not a string, so comparing it to the keys of `this` will not work as intended. You should directly invoke the callback with `callback.call(this, number);` without this check.